foreshadow/core/data_preparer.py

```diff
@@ -0,0 +1,66 @@
+"""Data preparation and foreshadow pipeline."""
+from sklearn.pipeline import Pipeline
+
+from foreshadow.cleaners.data_cleaner import DataCleaner as _DataCleaner
+
+
+def _none_to_dict(**kwargs):
+    """Transform input kwarg to valid dict, handling sentinel value.
+
+    Accepts a single kwarg by default.
+
+    Args:
+        **kwargs: the kwarg value to ensure is proper format for kwargs.
+
+    Returns:
+        kwarg set to default
+
+    Raises:
+        ValueError: if value of kwarg is not a valid value for kwarg (dict,
+            None). Also if > 1 kwargs passed.
+
+    """
+    if len(kwargs) > 1:
+        raise ValueError("Pass only 1 kwarg at a time.")
+    for key, val in kwargs.items():
+        val = {} if val is None else val
+        if not isinstance(val, dict):
+            raise ValueError(
+                "value for kwarg: {} must be dict or " "None.".format(key)
+            )
+        return val
+
+
+class DataPreparer(Pipeline):
+    """Predefined pipeline for the foreshadow workflow."""
+
+    def __init__(
+        self,
+        cleaner_kwargs=None,
+        intent_kwargs=None,
+        engineerer_kwargs=None,
+        preprocessor_kwargs=None,
+        reducer_kwargs=None,
+        modeler_kwargs=None,
```

Comment:
There wouldn't be a modeler here, it is a part of the Foreshadow object

Reply:
Will remove - you're right